rotkehlchen/api/v1/schemas.py

```diff
@@ -2622,3 +2625,67 @@ def make_user_notes_query(  # pylint: disable=no-self-use
         return {
             'filter_query': filter_query,
         }
+
+
+class CustomAssetsQuerySchema(DBPaginationSchema, DBOrderBySchema):
+    name = fields.String(load_default=None)
+    identifier = fields.String(load_default=None)
+    custom_asset_type = fields.String(load_default=None)
+
+    @post_load
+    def make_custom_assets_query(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> Dict[str, Any]:
+        filter_query = CustomAssetsFilterQuery.make(
+            limit=data['limit'],
+            offset=data['offset'],
+            order_by_rules=create_order_by_rules_list(
+                data=data,
+                default_order_by_field='name',
+                is_ascending_by_default=True,
+            ),
+            name=data['name'],
+            identifier=data['identifier'],
+            custom_asset_type=data['custom_asset_type'],
+        )
+        return {'filter_query': filter_query}
+
+
+class BaseCustomAssetSchema(Schema):
+    name = fields.String(required=True)
+    notes = fields.String(load_default=None)
+    custom_asset_type = fields.String(required=True)
+
+    @post_load
+    def make_custom_asset(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> Dict[str, CustomAsset]:
+        custom_asset = CustomAsset.initialize(
+            identifier=str(uuid4()),
+            name=data['name'],
+            notes=data['notes'],
+            custom_asset_type=data['custom_asset_type'],
+        )
+        return {'custom_asset': custom_asset}
+
+
+class EditCustomAssetSchema(BaseCustomAssetSchema):
+    identifier = fields.UUID(required=True)
```

Comment:
why do you need to specify uuid here? Would just string not be the same here?

Reply:
The reason for specifying `fields.UUID()` is that the identifier we generate for custom assets is UUID.

Reply:
and what's the benefit? I don't nderstand. What does `fields.UUID()` offer for us in our case over a simple string?